src/ajax.js

```diff
@@ -597,15 +598,16 @@ jQuery.extend( {
 
 			// If data is available, append data to url
 			if ( s.data ) {
-				cacheURL += ( rquery.test( cacheURL ) ? "&" : "?" ) + s.data;
+				cacheURL = cacheURL.replace( remptyQuery, "" ) +
+					( rquery.test( cacheURL ) ? "&" : "?" ) + s.data;
 
 				// #9682: remove data so that it's not used in an eventual retry
 				delete s.data;
 			}
 
-			// Add anti-cache in uncached url if needed
+			// Add or update anti-cache param if needed
 			if ( s.cache === false ) {
-				cacheURL = cacheURL.replace( rts, "" );
+				cacheURL = cacheURL.replace( rantiCache, "" ).replace( remptyQuery, "" );
```

Comment:
This code still exhibits the issue (e.g., it converts `/path?_=1&foo=bar` into `/pathfoo=bar`). The root cause of gh-3229 isn't a bad regex, it's a missing "$1" replacement string dropped by https://github.com/jquery/jquery/commit/e077ffb083743f4a4b990f586c9d25d787e7b417#diff-c3749d3acba09ca9ec16bb56e496408bL599 that we just never noticed.